core/store/src/opener.rs

```diff
@@ -569,3 +570,48 @@ pub trait StoreMigrator {
     /// equal to [`DB_VERSION`].
     fn migrate(&self, store: &Store, version: DbVersion) -> anyhow::Result<()>;
 }
+
+/// Creates checkpoint of hot storage in `home_dir.join(checkpoint_relative_path)`
+///
+/// If `columns_to_keep` is None doesn't cleanup columns.
+/// Otherwise deletes all columns that are not in `columns_to_keep`.
+///
+/// Returns NodeStorage of checkpoint db.
+///
+/// `config` -- StoreConfig for hot storage (needed to open checkpoint).
+/// `archive` -- is hot storage archival (needed to open checkpoint).
+#[allow(dead_code)]
+pub fn checkpoint_hot_storage_and_cleanup_columns(
+    db_storage: &NodeStorage,
+    home_dir: &std::path::Path,
+    checkpoint_relative_path: std::path::PathBuf,
+    columns_to_keep: Option<Vec<DBCol>>,
+    mut config: StoreConfig,
```

Comment:
Maybe this argument isn't needed. `StoreConfig::path` is the only thing important for the DB being opened as we're not using it to run any transactions (yet).